### DISCUSSION TITLE
How should I adjust the PVC size?

### DISCUSSION THREAD

**Discussion by skyshaokai** [openkruise/kruise#1119] [Q&A]
Cloneset  use volumeClaimTemplates when i modify storage from 10240Mi to 15360Mi found 1 error：
cannot patch object: admission webhook "vcloneset.kb.io" denied the request: spec: Forbidden: updates to cloneset spec for fields other than ''replicas'', ''template'', ''lifecycle'', ''scaleStrategy'', ''updateStrategy'',  ''minReadySeconds'' and ''revisionHistoryLimit'' are forbidden.

**Comment by veophi** [comment]
@skyshaokai   Currently (now is v1.3.0), `volumeClaimTemplates` of CloneSet is not allowed to mutate,  but we are planing to open this mutation operation.   This feature can be used in v1.3.1 as soon as possible.

**Reply by skyshaokai** [reply]
Does the plan support modifying the storage size and increasing or decreasing the volumeClaimTemplate? In addition, when is the function scheduled to be released?

**Reply by veophi** [reply]
> If I don't use volumeClaimTemplates, how should I use pvc for cloneset?

I think it will be hard to do that if you do not use volumeClaimTemplates.

This PR will allow you to mutate this field. But, for the change to take effect,  you must recreate your pods.
https://github.com/openkruise/kruise/pull/1118#issuecomment-1307107690

**Reply by skyshaokai** [reply]
Does the plan support modifying the storage size and increasing or decreasing the volumeClaimTemplate? In addition, when is the function scheduled to be released?

Do you mean that cloneset only supports modifying the volumeClaimTemplates field. However, if we want the pod to take effect, we need to trigger a pod reconstruction. For example, modify an env environment variable while modifying volumeClaimTemplates at same time.

**Reply by veophi** [reply]
> Does the plan support modifying the storage size and increasing or decreasing the volumeClaimTemplate? In addition, when is the function scheduled to be released?
> 
> Do you mean that cloneset only supports modifying the volumeClaimTemplates field. However, if we want the pod to take effect, we need to trigger a pod reconstruction. For example, modify an env environment variable while modifying volumeClaimTemplates at same time.

1. You have to reconstruct your pods to make this changes to take effect with this PR (https://github.com/openkruise/kruise/pull/1118#issuecomment-1307107690).
2.  We did have a plan to let you do this automatically.

**Reply by skyshaokai** [reply]
When is the function scheduled to be released?
